### PR TITLE
Remaining migration polishing

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- add telemetry
- backup files before overriding them
- add validations if we actually need to migrate stuff
- add more prints

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/cli/data.py [hunk 1]
```diff
@@ -410,27 +418,35 @@ def _migrate_model_config(args: argparse.Namespace) -> None:
...
-    # 3. Auto backup file?
     (
         model_configuration,
         domain,
         new_rules,
     ) = rasa.core.config.migrate_mapping_policy_to_rules(model_configuration, domain)
 
+    if new_rules:
+        _backup(domain_file)
+        domain.persist_clean(domain_file)
```
Comment (m-vdb) [inline rasa/cli/data.py:429]: why not do this at the end with the config?

File: rasa/cli/data.py [hunk 2]
```diff
@@ -410,27 +418,35 @@ def _migrate_model_config(args: argparse.Namespace) -> None:
...
     if new_rules:
+        _backup(configuration_file)
+        rasa.shared.utils.io.write_yaml(model_configuration, configuration_file)
         _dump_rules(rule_output_file, new_rules)
 
+    telemetry.track_data_convert("yaml", "config")
+
+    rasa.shared.utils.cli.print_success(
+        f"Finished migrating your policy configuration 🎉.\n"
+        f"The migration generated {len(new_rules)} which were added to "
```
Comment (m-vdb) [inline rasa/cli/data.py:446]:
```suggestion
        f"The migration generated {len(new_rules)} rule(s) which were added to "
```

Comment (m-vdb) [reply rasa/cli/data.py:446]: but ideally we would pluralise the "(s)" and the "was/were"

File: rasa/cli/data.py [hunk 3]
```diff
@@ -410,27 +418,35 @@ def _migrate_model_config(args: argparse.Namespace) -> None:
...
+        f"'{rule_output_file}'"
```
Comment (m-vdb) [inline rasa/cli/data.py:447]:
```suggestion
        f"'{rule_output_file}'."
```

**Review by m-vdb — APPROVED** [review]
Looks good!

File: rasa/cli/data.py (hunk 1 above)
Comment (wochinge) [reply rasa/cli/data.py:429]: this saved me one additional check 😀Added the check and moved this to the end of the function so all dumping things are at the end.

**Pull request merged**